src/persist-types/src/stats.rs

```diff
@@ -225,51 +205,6 @@ impl StructStats {
             )),
         }
     }
-
-    /// Trims the included column status until they fit within a budget.
-    ///
-    /// This might remove stats for a column entirely, unless `force_keep_col`
-    /// returns true for that column. The resulting StructStats object is
-    /// guaranteed to fit within the passed budget, except when the columns that
-    /// are force-kept are collectively larger than the budget.
-    pub fn trim_to_budget<F: Fn(&str) -> bool>(&mut self, budget: usize, force_keep_col: F) {
-        // Not trimming necessary should be the overwhelming common case in
-        // practice.
-        if self.cost() <= budget {
-            return;
-        }
-
-        // First try any lossy trimming that doesn't lose an entire column.
-        self.trim();
-        if self.cost() <= budget {
-            return;
-        }
-
-        // That wasn't enough. Sort the columns in order of ascending size and
-        // keep however many fit within the budget. This strategy both keeps the
-        // largest total number of columns and also optimizes for the sort of
-        // columns we expect to need stats in practice (timestamps are numbers
-        // or small strings).
```

Comment:
I think this comment was incorrect: the `Reverse` means we sort in _descending` order, so we keep the largest columns and discard the small ones.

Along with the other changes, the rewritten logic should remove the largest columns first. Let me know if you think this deserves a dedicated test!

Reply:
seems worth a test if it's straightforward